src/com/google/cose/Ec2SigningKey.java

```diff
@@ -220,4 +253,59 @@ public static Ec2SigningKey parse(byte[] keyBytes) throws CborException, CoseExc
   public static Ec2SigningKey decode(DataItem cborKey) throws CborException, CoseException {
     return new Ec2SigningKey(cborKey);
   }
+
+  public byte[] sign(Algorithm algorithm, byte[] message, String provider) throws CoseException {
```

Comment:
`@Nullable` would need to be updated everywhere if it is updated here since otherwise it could create confusion for non-annotated parameters. Hence, I am avoiding this and `@Nonnull` for the time being.